src/components/ViewPackagePage/components/package-header.tsx

```diff
@@ -40,7 +41,8 @@ export default function PackageHeader({
     packageInfo?.owner_github_username ===
     useGlobalStore((s) => s.session?.github_username)
   const isLoggedIn = useGlobalStore((s) => s.session != null)
-  const { OrderDialog, isOpen, open, close, stage, setStage } = useOrderDialog()
+  const signIn = useSignIn()
+  const { OrderDialog, isOpen, open, close, stage, setStage } = useOrderDialog(signIn)
```

Comment:
This input parameter API is very unclear, when the parameter doesn't logically follow (in this case `useOrderDialog` you would expect maybe `useOrderDialog(order)` not `useOrderDialog(signIn)`) so you have to use named parameters.

Whenever you pass a function to a hook that acts on user input, use the React `on*` convention

```suggestion
  const { OrderDialog, isOpen, open, close, stage, setStage } = useOrderDialog({
    onSignIn: signIn
  })
```

Reply:
also you need to pass a boolean indicating whether or not the user is signed in I think. `{ onSignIn, isLoggedIn }`- idk how you know a user is signed in (others will also be confused!)